modules/web/src/app/node-data/basic/provider/vmware-cloud-director/component.ts

```diff
@@ -392,5 +393,25 @@ export class VMwareCloudDirectorBasicNodeDataComponent
         [Controls.Catalog]: this.form.get(Controls.Catalog).value?.[ComboboxControls.Select],
       } as VMwareCloudDirectorNodeSpec,
     };
+
+    if (
+      !this._nodeDataService.isInWizardMode() &&
+      !this._initialQuotaCalculationPayload &&
+      !!this._nodeDataService.nodeData.creationTimestamp
+    ) {
+      this._initialQuotaCalculationPayload = {
+        ...payload,
+        [Controls.DiskSizeGB]: this.form.get(Controls.DiskSizeGB).value,
```

Comment:
We should not put this here but instead if we need it, then this should be added inside `payload` above.
Same comment for other providers as well.